### PR TITLE
fix(date-picker): no longer hides year for zh-CN locale

### PR DESCRIPTION
**Pull request by anveshmekala**
**Related Issue:** #5258 

This fix will unhide the year in date-picker component introduced as regression from `beta.84` version

A047343F-4A2D-4783-BEFA-52BE59C105D7_4_5005_c

### CODE REVIEW

File: src/components/date-picker-month-header/date-picker-month-header.tsx [hunk 1]
```diff
@@ -152,14 +151,7 @@ export class DatePickerMonthHeader {
               type="text"
               value={localizedYear}
             />
-            {suffix && (
-              <span class="suffix">
-                <span aria-hidden="true" class="suffix__invisible">
```
Comment (anveshmekala) [inline src/components/date-picker-month-header/date-picker-month-header.tsx:157]: Wasn't sure why we have the `span` element with `visibility: hidden`

File: src/components/date-picker-month-header/date-picker-month-header.scss [hunk 2]
```diff
@@ -117,22 +117,16 @@
...
+  @apply top-0
     left-0
-    w-16
     whitespace-nowrap
-    text-left;
-}
-
-.suffix__invisible {
-  @apply invisible;
+    text-right;
```
Comment (driskull) [inline src/components/date-picker-month-header/date-picker-month-header.scss:131]: Should this be `text-right` or something that flips in RTL?

File: src/components/date-picker-month-header/date-picker-month-header.scss [hunk 3]
```diff
@@ -117,22 +117,16 @@
 }
 
 .year--suffix {
-  @apply w-16 text-left;
+  @apply text-left;
```
Comment (driskull) [inline src/components/date-picker-month-header/date-picker-month-header.scss:120]: Should this flip in RTL?

File: src/components/date-picker-month-header/date-picker-month-header.tsx (hunk 1 above)
Comment (driskull) [reply src/components/date-picker-month-header/date-picker-month-header.tsx:157]: Looks like this span was here to hide it from screen readers. Seems like that functionality is removed here. Are we sure we can no longer hide this text from screen readers?

**Review by driskull — COMMENTED** [review]
Looking good @anveshmekala . Just had some questions/comments!

File: src/components/date-picker-month-header/date-picker-month-header.scss (hunk 3 above)
Comment (anveshmekala) [reply src/components/date-picker-month-header/date-picker-month-header.scss:120]: nope, it doesn't need to flip.

File: src/components/date-picker-month-header/date-picker-month-header.scss (hunk 2 above)
Comment (anveshmekala) [reply src/components/date-picker-month-header/date-picker-month-header.scss:131]: same here,  it doesn't need to flip. The order has to respect the direction.

File: src/components/date-picker-month-header/date-picker-month-header.tsx (hunk 1 above)
Comment (anveshmekala) [reply src/components/date-picker-month-header/date-picker-month-header.tsx:157]: The `span` element here is masking the `input` element which displays the year. If we want to hide the input element then this needs to be restored. Might be missing some context here.  @geospatialem  WDYT?

Comment (geospatialem) [reply src/components/date-picker-month-header/date-picker-month-header.tsx:157]: I'm hearing some invalid entries in the demos with NVDA. Wondering if the `aria-hidden` prop is used to hide those entries? Without this change, its registering as blank in the Arabic demos.

Comment (geospatialem) [reply src/components/date-picker-month-header/date-picker-month-header.tsx:157]: The comment above is when the screen reader is set to English. Once NVDA's language is set to the appropriate locale, the year number, or "year" text is read back to the screen reader. Great work, @anveshmekala! 👍🏻

**Comment by anveshmekala** [comment]
Adding @macandcheese to review the changes as we drop the `span` element which masks the `input` field. This change moves the suffix of the year by `6px` towards the right. 

OG version before the bug :
188026997-bad100c6-2a2b-47a8-b4ae-b56de76bba7f

After the fix :
A047343F-4A2D-4783-BEFA-52BE59C105D7_4_5005_c

cc: @ashetland

**Review by jcfranco — APPROVED** [review]
Code changes LGTM!

**Pull request merged**